modules/monitoring-configuring-pod-topology-spread-constraints.adoc

```diff
@@ -0,0 +1,180 @@
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      topologySpreadConstraints:
+      - maxSkew: 1
+        topologyKey: monitoring
+        whenUnsatisfiable: DoNotSchedule
+        labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: prometheus
+----
+
+. Save the file to apply the changes automatically.
++
+[WARNING]
+====
+When you save changes to the `cluster-monitoring-config` config map, the pods and other resources in the `openshift-monitoring` project might be redeployed.
+The running monitoring processes in that project might also restart.
+====
+
+* *To configure pod topology spread constraints for user-defined monitoring:*
+endif::openshift-dedicated,openshift-rosa[]
+
+. Edit the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
+----
+
+. Add the following settings under the `data/config.yaml` field to configure pod topology spread constraints:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    <component>: # <1>
+      topologySpreadConstraints:
+      - maxSkew: <n> # <2>
+        topologyKey: <key> # <3>
+        whenUnsatisfiable: <value> # <4>
+        labelSelector: # <5>
+          <match-option>
+----
+<1> Specify a name of the component for which you want to set up pod topology spread constraints.
+<2> Specify a numeric value for `maxSkew`, which defines the degree to which pods are allowed to be unevenly distributed.
+<3> Specify a key of node labels for `topologyKey`.
+Nodes that have a label with this key and identical values are considered to be in the same topology.
+The scheduler tries to put a balanced number of pods into each domain.
+<4> Specify a value for `whenUnsatisfiable`.
+Available options are `DoNotSchedule` and `ScheduleAnyway`.
+Specify `DoNotSchedule` if you want the `maxSkew` value to define the maximum difference allowed between the number of matching pods in the target topology and the global minimum.
+Specify `ScheduleAnyway` if you want the scheduler to still schedule the pod but to give higher priority to nodes that might reduce the skew.
+<5> Specify `labelSelector` to find matching pods. 
+Pods that match this label selector are counted to determine the number of pods in their corresponding topology domain.
++
+.Example configuration for Thanos Ruler
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    thanosRuler:
+      topologySpreadConstraints:
+      - maxSkew: 1
+        topologyKey: monitoring
+        whenUnsatisfiable: ScheduleAnyway
+        labelSelector:
+          matchLabels:
```

Comment:
matchLabels included here but missing in the example YAML above.

Reply:
Explained in the previous comment